src/Analyzers/CSharp/CodeFixes/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.cs

```diff
@@ -82,10 +82,27 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 }
 
                 var switchStatement = (SwitchStatementSyntax)switchLocation.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+                // If the swith statement expression is being implicitly converted then we need to explicitly cast the expression
+                // before rewriting as a switch expression
+                var expressionISymbolType = semanticModel.GetSymbolInfo(switchStatement.Expression).Symbol.GetSymbolType();
+                var expConversionISymbolType = semanticModel.GetTypeInfo(switchStatement.Expression).ConvertedType;
+
+                var needsCast = false;
+                SwitchStatementSyntax newSwitch = null;
+                if (expConversionISymbolType != expressionISymbolType && expConversionISymbolType != null)
```

Comment:
I would place the null check before the symbol comparison. You should also use `!SymbolEqualityComparer.Default.Equals(expConversionISymbolType, expressionISymbolType)` to compare the symbols I think